python/cog/server/worker.py

```diff
@@ -90,21 +212,26 @@ def terminate(self) -> None:
         if self._state == WorkerState.DEFUNCT:
             return
 
-        self._terminating = True
+        self._terminating.set()
         self._state = WorkerState.DEFUNCT
 
         if self._child.is_alive():
             self._child.terminate()
             self._child.join()
         self._events.shutdown()
+        if self._read_events_task:
+            self._read_events_task.cancel()
 
+    # FIXME: this will need to use a combination
+    # of signals and Cancel events on the pipe
     def cancel(self) -> None:
```

Comment:
Does it even make sense for `cancel` to be a method of `Worker` any more?

Reply:
We discussed this and the answer is "no but that's out of scope for this PR" which I think is fine.